docs/algorithms.md

```diff
@@ -0,0 +1,168 @@
+
+------
```

Comment:
1. Add link in readme, in design region.
2. Whether to describe separately according to the algorithm type？